src/botPage/view/blockly/blocks/trade/index.js

```diff
@@ -58,20 +58,25 @@ const replaceInitializationBlocks = (trade, ev) => {
     }
 };
 
+const setDefaultFields = (trade, parentFieldName, childFieldName) => {
+    const defaultValue = getDependentDropdownCallback[parentFieldName](trade)()[0][1];
+    trade.setFieldValue(defaultValue, childFieldName);
+};
+
 const resetTradeFields = (trade, ev) => {
     if (ev.blockId === trade.id) {
         if (ev.element === 'field') {
             if (ev.name === 'MARKET_LIST') {
```

Comment:
In here we could use the `setChildDefaultValue ` to set the dependent fields:

```
setChildDefaultValue(trade, ev.name);
```

This way we don't have to check the `ev.name` because `setChildDefaultValue` only sets the known ones.